Sources/Parameters/hyps_mask.h

```diff
@@ -0,0 +1,26 @@
+#ifndef HYP_MASK_H
+#define HYP_MASK_H
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+class HypsMask {
+public:
+  HypsMask();
+
+  // Base attributes on the as_dict method of parameter_helper.py
+  int nspecie = 1, ntwobody = 1, nthreebody = 1, nmanybody = 0, twobody_start,
+      threebody_start, manybody_start, ncut3b = 1;
+  std::vector<int> specie_mask, twobody_mask, threebody_mask, manybody_mask,
```

Comment:
I'm not sure I understand how this work. Here's a few observations and thoughts:

- The attributes look slightly different from the python hyps_mask we have. Kernel_name should be an array of strings
- Is there other places that call the hyps_mask content directly by name? In the python one, we can add all different types of mask if there are a new kernel type. As long as the kernel type is declared in Parameters.all_kernel_types, n_kernel_parameters. How do we add new attributes if we have new kernels? 
- We may need to change the default value of ntwobody, ..., nmanybody as -1 to void confusion if we have to hard coded all the attributes in the header.
- if everything has to be hard coded, we should write a test to check the consistency between python interface and the C code...

Reply:
Thanks for the comments! I needed to implement a C++ version of the hyperparameter mask in order to integrate my C++ environment class with the Python version. This is the foundation for all of the many-body kernels that I'm trying to integrate into the main repo ASAP. For that reason, I made the HypsMask class as simple as possible, and I agree that in the future, we should think carefully about how to design a proper class that works with any combination of kernels.

To address your specific concerns:
- The attributes should precisely match the Python hyps_mask, as it exists right now on the tip of the development branch. The kernel names are given in C++ as a vector of strings, which in Python translates to a list of strings.
- The flexibility of the Python version of the mask comes from the fact that it's implemented as a dictionary, not a class, which means that any keyword can be added to it, with no restrictions on the types of the corresponding values. This is flexible but chaotic, and makes it difficult to understand what the mask is doing. We should think about how to implement it in an object-oriented style, while still allowing multiple types of kernels to be accommodated. This should be addressed in the future, but is outside the scope of this PR.
- Instead of assigning an arbitrary default value, it would be better to have ntwobody, nmanybody, etc. directly linked to the corresponding masks, so that when the mask changes, the number changes automatically. This can be done by implementing them as properties of the mask class. We should certainly consider doing this in the future.
- I only hard coded things because that was the fastest way to proceed, but it's certainly not essential. Note that the C++ and Python versions of the mask are consistent enough for all the unit tests to pass on this branch, which should give some confidence, but we can of course consider adding additional tests in the future.

Reply:
Cool. How about opening a project on Github to list all the features to be implemented in C++? It's easier to keep track of and manage the changes via multiple pull requests.

Reply:
Good idea, I just opened up a project.